libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp

```diff
@@ -1315,12 +1321,46 @@ void ofGLProgrammableRenderer::bind(const ofFbo & fbo, bool setupPerspective){
 }
 
 //----------------------------------------------------------
-void ofGLProgrammableRenderer::unbind(const ofFbo & fbo){
-	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+void ofGLProgrammableRenderer::end(const ofFbo & fbo){
+	fbo.unbind();
```

Comment:
So if i understand correctly, you are talking about thread-safety and a race-condition around accessing `ofGetCurrentRenderer()` on the CPU (openFrameworks) side, and not on the GPU/OpenGL side?

I think i could find a way to address that.


---


Just to make sure: we are not talking about the implications of multiple contexts, each on its own thread, sharing resources, hopefully. Because this would be where things would get really, really complex, and i'm not sure oF can handle it 100% currently without introducing sync objects, mutexes and the like =) 

Also, it promises to be super slow (http://blog.gvnott.com/some-usefull-facts-about-multipul-opengl-contexts/) because of possible per-command context swaps.

In such a scenario it appears FBOs would not be shareable, because they are Container objects, like VAOs (https://www.opengl.org/wiki/OpenGL_Object#Container_objects). But how even to document this - or flag this in code? Or make an ofFbo not shareable?

It's really hard to find good documentation on this, i trawled the specs and the OpenGL registry with very meagre results. The most promising thing i could find was this: http://www.seas.upenn.edu/~pcozzi/OpenGLInsights/OpenGLInsights-AsynchronousBufferTransfers.pdf and it didn't appear too enthusiastic.

Sorry if i seem a bit confused, but i've been rewriting, and banging my head against this for 4 hours now, without finding a solid solution, and i'm a bit scared about the maintenance implications of `ofGLFWWindowSettings.shareContextWith = mainWindow;` to be honest =)